src/AnalysisServices/AnalysisServices.Dataplane/Models/AsAzureDataplaneClient.cs

```diff
@@ -0,0 +1,231 @@
+{
+    /// <summary>
+    /// Contains properties and logic for <see cref="AsAzureDataplaneClient"/>.
+    /// All permutations of CallGetAsync, and CallPostAsync are in the partial class in order to make this partial class more readable.
+    /// </summary>
+    /// <remarks>Most modifications will likely be in this part of the class.</remarks>
+    public partial class AsAzureDataplaneClient : ServiceClient<AsAzureDataplaneClient>, IAsAzureHttpClient
+    {
+        /// <summary>
+        /// The base Uri of the service.
+        /// </summary>
+        public Uri BaseUri { get; set; }
+
+        /// <summary>
+        /// Credentials needed for the client to connect to Azure.
+        /// </summary>
+        public ServiceClientCredentials Credentials { get; private set; }
+
+        private Func<HttpClient> HttpClientProvider { get; set; }
+
+        // This may cause problems, but is necessary for mulitple inheritance.
+        public new HttpClient HttpClient { get; set; }
+
+        public AsAzureDataplaneClient(Uri baseUri, ServiceClientCredentials credentials, Func<HttpClient> httpClientProvider, params DelegatingHandler[] handlers) : base(handlers)
+        {
+            this.BaseUri = baseUri ?? throw new ArgumentNullException(nameof(baseUri));
+            this.Credentials = credentials ?? throw new ArgumentNullException(nameof(credentials));
+            this.HttpClientProvider = httpClientProvider ?? throw new ArgumentNullException(nameof(httpClientProvider));
+            this.Credentials.InitializeServiceClient(this);
+            this.resetHttpClient();
+        }
+
+        public AsAzureDataplaneClient(Uri baseUri, ServiceClientCredentials credentials, Func<HttpClient> httpClientProvider, HttpClientHandler rootHandler, params DelegatingHandler[] handlers) : base(rootHandler, handlers)
+        {
+            this.BaseUri = baseUri ?? throw new ArgumentNullException(nameof(baseUri));
+            this.Credentials = credentials ?? throw new ArgumentNullException(nameof(credentials));
+            this.HttpClientProvider = httpClientProvider ?? throw new ArgumentNullException(nameof(httpClientProvider));
+            this.Credentials.InitializeServiceClient(this);
+            this.resetHttpClient();
+        }
+
+        public void resetHttpClient()
+        {
+            this.HttpClient = this.HttpClientProvider();
+        }
+
+        #region GetMethodAsyncOverloads
+
+        /// <summary>
+        /// Calls SendRequestAsync() for a GET using a blank correlationId.
+        /// </summary>
+        /// <param name="baseUri">The base Uri to call.</param>
+        /// <param name="requestUrl">The request Url.</param>
+        /// <param name="accessToken">The access token (will be discarded).</param>
+        /// <returns>The http response message.</returns>
+        public async Task<HttpResponseMessage> CallGetAsync(Uri baseUri, string requestUrl, string accessToken)
+        {
```

Comment:
do we need this one? the one below seems enough

Reply:
These are required methods in the implementation of IAsAzureHttpClient interface.